module/zstd/zstd.c

```diff
@@ -279,55 +303,10 @@ zstd_cookie_to_enum(int32_t level)
 		elevel = level;
 		return (elevel);
 	} else if (level < 0) {
-		switch (level) {
-			case -1:
-				return (ZIO_ZSTDLVL_FAST_1);
-			case -2:
-				return (ZIO_ZSTDLVL_FAST_2);
-			case -3:
-				return (ZIO_ZSTDLVL_FAST_3);
-			case -4:
-				return (ZIO_ZSTDLVL_FAST_4);
-			case -5:
-				return (ZIO_ZSTDLVL_FAST_5);
-			case -6:
-				return (ZIO_ZSTDLVL_FAST_6);
-			case -7:
-				return (ZIO_ZSTDLVL_FAST_7);
-			case -8:
-				return (ZIO_ZSTDLVL_FAST_8);
-			case -9:
-				return (ZIO_ZSTDLVL_FAST_9);
-			case -10:
-				return (ZIO_ZSTDLVL_FAST_10);
-			case -20:
-				return (ZIO_ZSTDLVL_FAST_20);
-			case -30:
-				return (ZIO_ZSTDLVL_FAST_30);
-			case -40:
-				return (ZIO_ZSTDLVL_FAST_40);
-			case -50:
-				return (ZIO_ZSTDLVL_FAST_50);
-			case -60:
-				return (ZIO_ZSTDLVL_FAST_60);
-			case -70:
-				return (ZIO_ZSTDLVL_FAST_70);
-			case -80:
-				return (ZIO_ZSTDLVL_FAST_80);
-			case -90:
-				return (ZIO_ZSTDLVL_FAST_90);
-			case -100:
-				return (ZIO_ZSTDLVL_FAST_100);
-			case -500:
-				return (ZIO_ZSTDLVL_FAST_500);
-			case -1000:
-				return (ZIO_ZSTDLVL_FAST_1000);
-			default:
-				/* This shouldn't happen. Cause a panic. */
-			printk(KERN_ERR
-			    "%s:Invalid ZSTD level encountered: %d",
-			    __func__, level);
-			return (ZIO_ZSTD_LEVEL_DEFAULT);
+		int i;
+		for (i = 0; i < ARRAYSIZE(fastlevels); i++) {
```

Comment:
Isn't a for slower than a switch, somewhat?

Reply:
yes, but if you can measure the difference you must be a friend of werner heisenberg. that big mess of switch statement 2 times is a good reason to combine it. its also easier maintainable by just editing the value in the array itself